airflow/serialization/serde.py

```diff
@@ -57,11 +57,9 @@
 _extra_allowed: set[str] = set()
 
 _primitives = (int, bool, float, str)
-_iterables = (list, set, tuple)
+_collections = (list, set, tuple)  # dict is treated specially.
```

Comment:
Renamed to better reflect these classes. _Iterable_ has special meaning in Python (collection also does, but its meaning is more in-line with what we mean here).